stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -70,15 +70,18 @@ public void disable(JsonRpcPeer peer, JSONObject params) {
 
   @ChromeDevtoolsMethod
   public JsonRpcResult getDocument(JsonRpcPeer peer, JSONObject params) {
-    Object rootElement = mDOMProvider.getRootElement();
-    if (rootElement == null) {
-      return null;
-    }
-
-    Node rootNode = createNodeForElement(rootElement);
+    final GetDocumentResponse result = new GetDocumentResponse();
+
+    mDOMProvider.postAndWait(new Runnable() {
+       @Override
+       public void run() {
+         Object rootElement = mDOMProvider.getRootElement();
+         if (rootElement != null) {
+           result.root = createNodeForElement(rootElement);
```

Comment:
While this assignment is safe because `postAndWait` locks consistently on both threads, I'd prefer clarifying that by pattern as such:

```java
final AtomicReference<...> rootNode = new AtomicReference<>(...);
postAndWait(() -> {
  rootNode.set(theThing);
});
```